File: content-solutions/stack/widest-banner.js
function solution(heights) {
  const h = [...heights, 0];
  const stack = [];
  let best = 0;
  for (let i = 0; i < h.length; i++) {
    while (stack.length > 0 && h[stack[stack.length - 1]] >= h[i]) {
      const top = stack.pop();
      const left = stack.length > 0 ? stack[stack.length - 1] + 1 : 0;
      best = Math.max(best, h[top] * (i - left));
    }
    stack.push(i);
  }
  return best;
}

File: content-solutions/stack/widest-banner.java
import java.util.*;

class Solution {
    public int solution(int[] heights) {
        int n = heights.length;
        Deque<Integer> stack = new ArrayDeque<>();
        int best = 0;
        for (int i = 0; i <= n; i++) {
            int h = i == n ? 0 : heights[i];
            while (!stack.isEmpty() && heights[stack.peek()] >= h) {
                int top = stack.pop();
                int left = stack.isEmpty() ? 0 : stack.peek() + 1;
                best = Math.max(best, heights[top] * (i - left));
            }
            stack.push(i);
        }
        return best;
    }
}
